pipeline/transformer/transformer_sierra/src/main/scala/weco/pipeline/transformer/sierra/transformers/SierraGenres.scala

```diff
@@ -48,18 +48,21 @@ object SierraGenres
   def apply(bibData: SierraBibData): List[Genre[IdState.Unminted]] =
     bibData
       .varfieldsWithTag("655")
-      .map {
+      .flatMap {
         varField =>
           val (primarySubfields, subdivisionSubfields) =
             getLabelSubfields(varField)
 
-          val label = getLabel(primarySubfields, subdivisionSubfields)
-          val concepts = getPrimaryConcept(
-            primarySubfields,
-            varField = varField
-          ) ++ getSubdivisions(subdivisionSubfields)
+          getLabel(primarySubfields, subdivisionSubfields) match {
+            case "" => None
+            case label =>
+              val concepts = getPrimaryConcept(
+                primarySubfields,
+                varField = varField
+              ) ++ getSubdivisions(subdivisionSubfields)
```

Comment:
nit: I would keep the two variable assignments from the original code, then add the `match` block as a new thing – there's a lot going on in a couple of lines that make it a bit tricky to follow.

Reply:
Thanks, Alex! This is such a great learning point. 

Something like this?

```scala
val (primarySubfields, subdivisionSubfields) = getLabelSubfields(varField)
      val label = getLabel(primarySubfields, subdivisionSubfields)
      label match {
        case "" => None
        case _ =>
          val concepts = getPrimaryConcept(
            primarySubfields,
            varField = varField
          ) ++ getSubdivisions(subdivisionSubfields)
```

Reply:
I'd move the `concepts` variable out as well, so the `match` arms are both quite short. You'll only use the variable in one of the arms, but then you can glance over the `match` to see what each arm is doing.